packages/sdk/src/evm/core/classes/rpc-connection-handler.ts

```diff
@@ -85,37 +85,39 @@ export function getSignerAndProvider(
   let provider: providers.Provider | undefined;
 
   if (Signer.isSigner(network)) {
+    // Here, we have an ethers.Signer
     signer = network;
     if (network.provider) {
       provider = network.provider;
     }
+  } else if (providers.Provider.isProvider(network)) {
+    // Here, we have an ethers.providers.Provider
+    provider = network;
+  } else {
+    // Here, we must have a ChainOrRpcUrl, which is a chain name, chain id, rpc url, or chain config
+    // All of which, getChainProvider can handle for us
+    provider = getChainProvider(network, options);
   }
 
   if (options?.readonlySettings) {
-    provider = getReadOnlyProvider(
+    // If readonly settings are specified, then overwrite the provider
+    provider = getProviderFromRpcUrl(
```

Comment:
I worry that we're always overriding the provider here, even when a provider is passed explicitely.

Like if I pass a custom WalletConnect UniversalProvider here for example, you will break me by using a regular json rpc provider.

I think the previous logic ONLY used the read only provider if one wasn't passed explicitely. that's safer i think.

So this would be:
`if (options?.readonlySettings && !provider)`

Reply:
The previous logic always overwrote - but this suggestion is better I think

Reply:
@adam-maj hm you're right the previous logic always overwrote. I guess its fine to leave it as is. its controllable via sdk options anyways